enhancements/agent-installer/automated-workflow-kubeconfig.md

```diff
@@ -0,0 +1,253 @@
+  and booting the ISO on the hosts.
+
+### Goals
+
+* Generate a working admin kubeconfig as part of the `openshift-install agent
+  create image` command.
+
+### Non-Goals
+
+* Make the admin kubeconfig available earlier in the hosted assisted UI.
+
+## Proposal
+
+The admin kubeconfig will be generated by the `agent create image` command
+using an Asset based on the existing `kubeconfig.AdminClient` one. This will
+differ only in that it will get the cluster API address from the agent
+cluster-manifests, rather than the install-config, because the install-config
+is optional in the agent-based installer flow.
+
+The actual ignition files used for bootstrapping the cluster are not created by
+the agent-based installer binary, but rather by the assisted-service running
+openshift-installer again. In order to ensure that the cluster that is
+installed accepts the credentials in the kubeconfig, we must supply all assets
+that match the following rules:
+
+If we define an 'implicated' asset as one that:
+
+* Contains some additional data that cannot be recreated from its dependencies alone (unlike, e.g. the CABundle assets); and
+* Can be written to and loaded from disk; and
+* Either:
+  * Is depended on (directly or indirectly) by the implicating asset; or
+  * Itself implicates one of the assets that would be implicated if it could be
+    loaded from disk, but is not.
+
+then we must supply all assets that are implicated by the
+`kubeconfig.AdminClient` to the ignition generation step of the installer run
+by assisted-service. This will ensure that the built cluster accepts the
+credentials in the pre-existing kubeconfig.
+
+The status quo is that amongst the writable assets (shown in color below),
+there are some (shown in red) that cannot actually be read back by the
+installer. As a result, both the nature and the sheer number of assets required
+would preclude this approach:
+
+![Assets implicated by kubeconfig (status quo)](./kubeconfig_implications_without_changes.svg)
+
+However, by adding the ability to read back a handful of assets (shown in blue
+below) that currently cannot be, this complexity is massively reduced:
+
+![Assets implicated by kubeconfig with cert/key loading](./kubeconfig_implications_with_cert_loading.svg)
+
+Accordingly, the installer will be modified to allow reading the following
+certificate assets from disk:
+
+* `tls.KubeAPIServerLBSignerCertKey`
+* `tls.KubeAPIServerLocalhostSignerCertKey`
+* `tls.KubeAPIServerServiceNetworkSignerCertKey`
+* `tls.AdminKubeConfigSignerCertKey`
+* `tls.AdminKubeConfigClientCertKey`
+
+When generating the agent ISO, the certificate and key files for those same
+assets will be built in to the ISO.
+
+The assisted-service will copy the certificates from a location provided by the
+environment variable `ASSISTED_SERVICE_CLUSTER_TLS_CERTS_DIR` to the
+installer's work directory before generating the manifests/ignition.
```

Comment:
Do we only want this bypass approach to apply to the auth-related manifests, rather than making it a generic way to use other manifests without passing them through the API?

Reply:
And we don't want the auth-related manifests to go through that same API?

I'm not arguing one way or the other, but it would be good to have that laid out explicitly with the reasoning.

Reply:
To be clear, the tls assets are _not_ manifests. They are not kubectl applied to the cluster, and they don't live in the `manifests` directory. They're just raw certificate and private key files.

Reply:
OK, thanks for clarifying that.